site-manage/pageorder/tool/src/java/org/sakaiproject/site/tool/helper/order/impl/SitePageEditHandler.java

```diff
@@ -584,9 +580,24 @@ private boolean pageVisibilityHelper(String pageId, boolean visible, boolean ena
             }
             
             if (saveChanges) {
-                if ( visibility != null ) {
-                    roleConfig.setProperty(PORTAL_VISIBLE, visibility);
+                final Boolean specialHidden = getSitePropertySpecialHidden();
+                if(Boolean.TRUE.equals(specialHidden) && "sakai.resources".equals(toolId)) {
+                    final String siteCollectionId =  contentHostingService.getSiteCollection(placement.getSiteId());
+                    // It is possible, though unlikely, that remove/add property will throw a InUseException or ServerOverloadException
+                    // if so do nothing - ?
+                    try{
+                        if (visibility.equals("true")) {
+                            contentHostingService.removeProperty(siteCollectionId, ResourceProperties.PROP_HIDDEN_WITH_ACCESSIBLE_CONTENT);
+                        } else {
+                            contentHostingService.addProperty(siteCollectionId, ResourceProperties.PROP_HIDDEN_WITH_ACCESSIBLE_CONTENT, "true");
+                        }
+                    } catch (InUseException | ServerOverloadException e) {
+                        // do nothing
```

Comment:
do nothing is fine but you should log it, maybe use level warn, and include the message of the exception in the log.